server/build_event_protocol/build_event_handler/build_event_handler.go

```diff
@@ -1034,6 +1048,82 @@ func LookupInvocation(env environment.Env, ctx context.Context, iid string) (*in
 	return invocation, nil
 }
 
+func GetCacheScoreCard(ctx context.Context, env environment.Env, req *capb.GetCacheScoreCardRequest) (*capb.GetCacheScoreCardResponse, error) {
+	page := &pgpb.OffsetLimit{Offset: 0, Limit: scoreCardPageSize}
+	if req.PageToken != "" {
+		var err error
+		page, err = paging.DecodeOffsetLimit(req.PageToken)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	scorecard, err := readScoreCard(ctx, env, req.InvocationId)
+	if err != nil {
+		return nil, err
+	}
+
+	end := page.Offset + page.Limit
+	scorecard.Results = scorecard.Results[int(page.Offset):end]
```

Comment:
you're validating the end value **after** subsetting the slice
it appears you are also not validating page.Offset index

Reply:
woops, messed this up when refactoring. Fixed